modules/clients/main.tf

```diff
@@ -0,0 +1,88 @@
+data "google_compute_subnetwork" "this" {
+  count   = length(var.subnets_list)
+  name    = var.subnets_list[count.index]
+  project = var.project_id
+  region  = var.region
+}
+
+locals {
+  private_nic_first_index = var.assign_public_ip ? 1 : 0
+  preparation_script      = templatefile("${path.module}/init.sh", {
+    yum_repo_server = var.yum_repo_server
+  })
+
+  mount_wekafs_script = templatefile("${path.module}/mount_wekafs.sh", {
+    all_subnets        = split("\n",replace(join("\n", data.google_compute_subnetwork.this.*.ip_cidr_range),"/\\S+//",""))[0]
+    all_gateways       = join(" ",data.google_compute_subnetwork.this.*.gateway_address)
+    nics_num           = var.nics
+    backend_ips        = join(" ", var.backend_ips)
+    mount_clients_dpdk = var.mount_clients_dpdk
+  })
+
+  custom_data_parts = [local.preparation_script, local.mount_wekafs_script]
+  vms_custom_data   = join("\n", local.custom_data_parts)
+}
+
+resource "google_compute_instance_template" "this" {
+  name           = "${var.prefix}-${var.cluster_name}-clients"
+  machine_type   = var.machine_type
+  can_ip_forward = false
+  project        = var.project_id
+  tags           = ["${var.prefix}-${var.cluster_name}-clients"]
+  labels         = {
+    weka_cluster_name = var.cluster_name
+  }
+  service_account {
+    email  = var.sa_email
+    scopes = ["cloud-platform"]
+  }
+  disk {
+    source_image = var.source_image
+    disk_size_gb = var.disk_size
+    boot         = true
+  }
+
+  # nic with public ip
+  dynamic "network_interface" {
+    for_each = range(local.private_nic_first_index)
+    content {
+      subnetwork = data.google_compute_subnetwork.this[network_interface.value].id
+      access_config {}
+    }
+  }
+
+  # nics with private ip
+  dynamic "network_interface" {
+    for_each = range(local.private_nic_first_index, var.nics)
+    content {
+      subnetwork = data.google_compute_subnetwork.this[network_interface.value].id
+    }
+  }
+
+  disk {
+    interface    = "NVME"
+    boot         = false
+    type         = "SCRATCH"
+    disk_type    = "local-ssd"
+    disk_size_gb = var.disk_size
+  }
+
+  lifecycle {
+    ignore_changes = [network_interface]
+  }
+
+  metadata_startup_script = local.vms_custom_data
+}
+
+# ======================== instance-group ============================
+resource "google_compute_instance_group_manager" "this" {
```

Comment:
no need for group manager, we use unmanaged group for backends, better to just use instances here